dev_docs/07_visual_testing_guide.md

```diff
@@ -26,123 +32,180 @@ KDS has a visual testing system that allows you to take snapshots of how KDS Com
+```html
+<template>
+  <VisualTestLayout>
+    <VisualTestExample
+      title="Label"
+      width="400px"
+      loadExample="KTextbox/WithLabel.vue"
+    />
 
-     - `renderComponentForVisualTest(component, props, slots)`: Renders the specified component with given props and slots in the visual testing playground.
-     - `takeSnapshot(name, options)`: Takes a Percy snapshot with the given name and options.
-     - `click(selector)`, `hover(selector)`, `scrollToPos(selector, scrollOptions)`, `waitFor(selector)`, `delay(time)`: Utility functions for simulating user interactions.
+    <VisualTestExample
+      title="Validation"
+      width="400px"
+      loadExample="KTextbox/Validation.vue"
+    />
 
-2. **Write Tests**:
+    ...
 
-   - Use the utility functions to render components and take snapshots. For example:
+  </VisualTestLayout>
+</template
+```
 
-     ```javascript
-     describe.visual('KButton Visual Tests', () => {
-       it('Sample test for KButton', async () => {
-         await renderComponentForVisualTest('KButton', {
-           text: 'Raised Button',
-           appearance: 'raised-button',
-         });
-         await takeSnapshot('KButton - Raised Button', { widths: [375, 520] });
-       });
-     });
-     ```
+4. **Register** `KTextboxVisualTest.vue` in [`/jest.conf/visual.load-test-components.js`](../jest.conf/visual.load-test-components.js).
+   
+```javascript
+import KTextboxVisualTest from '~~/lib/KTextbox/__tests__/components/KTextboxVisualTest.vue';
 
-    The components we can render with `renderComponentForVisualTest` are the ones that are already registered for the KDS Docs nuxt server, that means, the KDS components that are exposed in [KThemePlugin](../lib/KThemePlugin.js), and the docs common components that are registered in the [load-common-components.js file](../docs/plugins/load-common-components.js). If you need to use a component that is not registered here, you can refer to the **Example involving more complex component structures** point below.
+Vue.component('KTextboxVisualTest', KTextboxVisualTest);
+```
 
-     Note that the `widths` parameter passed to the `takeSnaphot` function is a part of Percy CLI's snapshot options. For a full list of available options, refer the [Percy documentation](https://www.browserstack.com/docs/percy/take-percy-snapshots/snapshots-via-scripts#per-snapshot-configuration).
+5. **Add `visual.spec.js`** for `KTextbox` ([`/lib/KTextbox/__tests__/visual.spec.js`](../lib/KTextbox/__tests__/visual.spec.js)). Use `renderComponentForVisualTest` and `takeSnapshot` to capture the `KTextboxVisualTest` snapshot. **Be sure to use `describe.visual` or `it.visual` instead of the default Jest notations.** For a full list of available snapshot options for `takeSnapshot`, refer to the [Percy documentation](https://www.browserstack.com/docs/percy/take-percy-snapshots/snapshots-via-scripts#per-snapshot-configuration).
```

Comment:
Alright, I see we are now having a different spec file to run visual tests as a convention, seems like the most robust option. This will solve #https://github.com/learningequality/kolibri-design-system/issues/1015!!!

(This also removes the need for having something like an it.visual, but it doesnt hurt to have it already set up 😅 )